Support pre-commit hooks.

arturkow2000 opened the pull request.

arturkow2000 committed:
Support pre-commit hooks.

krystian-hebel commented on entrypoint.sh:
```diff
@@ -19,6 +19,10 @@ else
 
+    # Shellcheck uses Docker. Docker socket may not be accessible due to
+    # permissions, so make docker binary setuid.
+    chmod +s "$(which docker)"
```
It is possible to skip adding setuid to keep permissions more constrained:
- read GID of docker.sock (e.g. `stat -c "%g" /var/run/docker.sock`)
- create group for that GID (only if not present), normally this should be `docker` group but GID for that group may be different in container than it is on host
- add `builder` user to that group

arturkow2000 replied:
Shellcheck is now installed in this container and we no longer access host's Docker.

arturkow2000 committed:
Add shellcheck to container.

Previous approach passed host Docker socket so that oficial shellcheck
hook could create sibling container to run shellcheck in, from now on we
use shellcheck provided by distro.
